editor/components/post-type-support-check/index.js

```diff
@@ -1,29 +1,37 @@
 /**
  * External dependencies
  */
-import { get, some, castArray } from 'lodash';
+import { get, includes, some, castArray } from 'lodash';
 
 /**
  * WordPress dependencies
  */
 import { withSelect } from '@wordpress/data';
 
-function PostTypeSupportCheck( { postType, children, supportKeys } ) {
+export function PostTypeSupportCheck( { postType, children, supportKeys, themeSupports } ) {
+	supportKeys = castArray( supportKeys );
 	const isSupported = some(
-		castArray( supportKeys ), ( key ) => get( postType, [ 'supports', key ], false )
+		supportKeys, ( key ) => get( postType, [ 'supports', key ], false )
 	);
 
 	if ( ! isSupported ) {
 		return null;
 	}
 
+	// 'thumbnail' and 'post-thumbnails' are intentionally different.
+	if ( includes( supportKeys, 'thumbnail' ) &&
```

Comment:
Why does `PostTypeSupportCheck` need to be handling this exception? If they're truly intentionally different, why lump them to be considered together? Should it be up to the consuming component to check for both the post type and theme supports? Notably, I see this as being part of the reason `PostFeaturedImageCheck` exists as a standalone component, so it can become:

```jsx
function PostFeaturedImageCheck( props ) {
	return (
		<ThemeSupportCheck supportKeys="post-thumbnails">
			<PostTypeSupportCheck { ...props } supportKeys="thumbnail" />
		</ThemeSupportCheck>
	);
}
```

Reply:
`ThemeSupportCheck` is a good suggestion. I'll open a new PR with a different implementation.